examples/radiation_transport/radiation_calculation.py

```diff
@@ -0,0 +1,194 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh, J. Morris,
+#                    D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+Example core and scraper-off layer radiation source calculation
+"""
+
+# %%
+import os
+
+import matplotlib.pyplot as plt
+
+import bluemira.codes.process as process
+from bluemira.base.config import Configuration
+from bluemira.base.file import get_bluemira_path
+from bluemira.base.parameter import ParameterFrame
+from bluemira.equilibria import Equilibrium
+from bluemira.geometry._deprecated_loop import Loop
+from bluemira.radiation_transport.advective_transport import ChargedParticleSolver
+from bluemira.radiation_transport.radiation_profile import STCore, STScrapeOffLayer
+
+# %%[markdown]
+# Read equilibrium
+
+# %%
+read_path = get_bluemira_path("equilibria", subfolder="data")
+eq_name = "DN-DEMO_eqref.json"
+eq_name = os.sep.join([read_path, eq_name])
+eq = Equilibrium.from_eqdsk(eq_name, load_large_file=True)
+
+# %%[markdown]
+# Get first wall shape
+
+# %%
+read_path = get_bluemira_path("radiation_transport/test_data", subfolder="tests")
+fw_name = "DN_fw_shape.json"
+fw_name = os.sep.join([read_path, fw_name])
+fw_shape = Loop.from_file(fw_name)
+
+# %%[markdown]
+# Run particle solver
+
+# %%
+p_solver_params = ParameterFrame()
+solver = ChargedParticleSolver(p_solver_params, eq, dx_mp=0.001)
```

Comment:
Seems to me like you just need the equilibrium and one other thing that the `ChargedParticleSolver` calculates (i.e. dx_imp and the flux surfaces). Why not just use the equilibrium and the flux surfaces?

Reply:
You mean as independent objects?
I call the ChargedParticleSolver to have the same set of flux surfaces